OpenBLAS 0.3.9 build segfault on ARM64 + 64-bit integer support

We've been observing this segfault for a while now, so it is not specific to `0.3.9`.

Invocation line: 
`make DYNAMIC_ARCH=1 USE_OPENMP=0 NUM_THREADS=64 BINARY=64 INTERFACE64=1 SYMBOLSUFFIX=64_ OBJCONV=/io/objconv/objconv`

Link to build/failure log: https://travis-ci.org/MacPython/openblas-libs/jobs/659863204?utm_medium=notification&utm_source=github_status

Summary of failure:

```
Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0xffff9b1509ab
#1  0xffff9b14fa1f
#2  0xffff9b2c65b7
#3  0xbe80d4
make[1]: *** [level1] Segmentation fault (core dumped)
make[1]: *** Waiting for unfinished jobs....
make: *** [tests] Error 2
```

Example PR exhibiting the segfault: https://github.com/MacPython/openblas-libs/pull/26

If you'd like me to try some things let me know. Thanks!

--- a/Makefile.system
+++ b/Makefile.system
@@ -635,6 +635,16 @@ endif
 ifeq ($(ARCH), arm64)
 NO_BINARY_MODE  = 1
 BINARY_DEFINED  = 1
+ifdef INTERFACE64
+ifneq ($(INTERFACE64), 0)
+ifeq ($(F_COMPILER), GFORTRAN)
+FCOMMON_OPT +=  -fdefault-integer-8
+endif
+ifeq ($(F_COMPILER), FLANG)
+FCOMMON_OPT += -i8
+endif
+endif
+endif
 endif
 
 

